File: src/domain/value-objects/PasswordPolicy.ts
/** Clave temporal fija para altas y restablecimientos. */
export const DEFAULT_TEMPORARY_PASSWORD = '87654321'

/** Claves temporales antiguas que tampoco se aceptan como definitivas. */
const LEGACY_TEMPORARY_PASSWORDS = new Set(['12345678', '87654321'])

/** Mínimo de Firebase Auth. El técnico puede elegir una clave sencilla. */
const MIN_PASSWORD_LENGTH = 6

export function isSecurePassword(password: string): boolean {
  if (password.length < MIN_PASSWORD_LENGTH) return false
  if (LEGACY_TEMPORARY_PASSWORDS.has(password)) return false
  return true
}

export function securePasswordRequirementsMessage(): string {
  return `Mínimo ${MIN_PASSWORD_LENGTH} caracteres. Puede ser sencilla. No uses la temporal ${DEFAULT_TEMPORARY_PASSWORD}.`
}

File: consorcio/lib/domain/value_objects/password_policy.dart
const defaultTemporaryPassword = '87654321';
const legacyTemporaryPasswords = {'12345678', '87654321'};
const minPasswordLength = 6;

bool isSecurePassword(String password) {
  if (password.length < minPasswordLength) return false;
  if (legacyTemporaryPasswords.contains(password)) return false;
  return true;
}

String securePasswordRequirementsMessage() {
  return 'Mínimo $minPasswordLength caracteres. Puede ser sencilla. No uses la temporal $defaultTemporaryPassword.';
}
